tests/commons.py

```diff
@@ -21,10 +27,18 @@ async def __anext__(self):
         return item
 
     def __call__(self, *args, **kwargs):
+        self.call_count += 1
+
         if args:
             self.call_args.append(args)
 
         if kwargs:
             self.call_kwargs.append(kwargs)
 
         return self
+
+    def assert_not_called(self):
+        return self.call_count == 0
+
+    def assert_called_once(self):
+        return self.call_count == 1
```

Comment:
I think this capability exists already: https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_once

Reply:
Yes but not on our custom `AsyncIterator` (which should probably be called `AsyncGenerator` class :) )